pkg/controller/cheinstallation/cheinstallation_controller.go

```diff
@@ -194,6 +225,80 @@ func (r *ReconcileCheInstallation) ensureCheSubscription(logger logr.Logger, che
 	return false, nil
 }
 
+// ensureWatchCheCluster adds watch for CheCluster resource if CheCluster CRD is installed else return requeue with true
+// CheCluster CRD may takes time to get installed until CheOperator is installed successfully
+// Once watch addded for CheCluster, sub-sequent calls to ensureWatchCheCluster() will do nothing
+func (r *ReconcileCheInstallation) ensureWatchCheCluster() (bool, error) {
+	r.mu.Lock()
+	defer r.mu.Unlock()
+	if r.watchCheCluster != nil {
+		cheClusterCRD := &apiextnv1beta1.CustomResourceDefinition{}
+		if err := r.client.Get(context.TODO(), types.NamespacedName{Name: "checlusters.org.eclipse.che"}, cheClusterCRD); err != nil {
+			if errors.IsNotFound(err) {
+				log.Info("requeue required", "message", err.Error())
+				return true, nil
+			}
+			return false, err
+		}
+		if err := r.watchCheCluster(); err != nil {
+			if meta.IsNoMatchError(err) {
```

Comment:
When can we have this case? Shouldn't we be safe if we already checked that the CRD exists?

Reply:
This is an extra safety guard and can be removed.  I have kept for extra safety and it can be removed.